server/src/main/java/org/opensearch/identity/noop/NoopTokenManager.java

```diff
@@ -26,8 +28,13 @@ public class NoopTokenManager implements TokenManager {
      * @return a new Noop Token
      */
     @Override
-    public AuthToken issueToken(String audience) {
+    public AuthToken issueOnBehalfOfToken(Map<String, Object> claims) {
```

Comment:
Lets try to make this interface as restrictive as possible.  I think we should require a subject as a parameter so it isn't captured as a side effect, and I think we should change the map into a POJO that clearly defines the claims so we can expect all of them to be defined or error out.

```suggestion
    public AuthToken issueOnBehalfOfToken(final Subject subject, final OnBehalfOfClaims claims) {
```

Note; I think its reasonable to check `if (subject != identity.getSubject()) { throw new NotSupportedException(...);} ` until / if broad impersonation is possible.

Reply:
@peternied Why pass in subject explicitly? Where would this method be called where the caller wants to pass in a subject other than the currently authenticated subject?

I asked @scrawfor99 to make the params generic (i.e. `Map<String, Object>`) to give the maximum flexibility to the implementer to determine what to do with the claims passed in, but I agree that it doesn't give the caller enough info on what's required for the call.

Reply:
Pure functions are easier to read and reduce the potential for errors due to misunderstandings.  This API is called `issueOnBehalfOfToken`, knowing 'who' the token is on behalf of is a critical part to understand how this function works.

Maximum flexibility also means expanding complexity, no?  I'd like to avoid complexity for this first iteration.